ROS1/catkin_ws/src/autonomous_vehicle_simulator/auto_vehicle/src/RoadDetector.cpp

```diff
@@ -0,0 +1,381 @@
+
+#include <string>
+#include <vector>
+#include "../include/RoadDetector.hpp"
+
+// IMAGE BLURRING
+/**
+ *@brief Apply gaussian filter to the input image to denoise it
+ *@param inputImage is the frame of a video in which the
+ *@param lane is going to be detected
+ *@return Blurred and denoised image
+ */
+cv::Mat RoadDetector::deNoise(cv::Mat inputImage)
+{
+    cv::Mat output;
+
+    cv::GaussianBlur(inputImage, output, cv::Size(3, 3), 0, 0);
+
+    return output;
+}
+
+// EDGE DETECTION
+/**
+ *@brief Detect all the edges in the blurred frame by filtering the image
+ *@param img_noise is the previously blurred frame
+ *@return Binary image with only the edges represented in white
+ */
+cv::Mat RoadDetector::edgeDetector(cv::Mat img_noise)
```

Comment:
```suggestion
cv::Mat RoadDetector::edgeDetector(cv::Mat img_noise) const
```

I don't think you are changing any class member, so `const` would be the way to go. Also try `static` if you are not reading any class member either